lms/djangoapps/courseware/user_state_client.py

```diff
@@ -253,7 +253,7 @@ def delete_many(self, username, block_keys, scope=Scope.user_state, fields=None)
         student_modules = self._get_student_modules(username, block_keys)
         for student_module, _ in student_modules:
             if fields is None:
-                student_module.state = None
+                student_module.state = "{}"
```

Comment:
If no state has ever been set, then `student_module.state` is `None`. If state has been set at some point, but has since been deleted, then `student_module.state` is `{}`.